spec/jobs/generate_hathifile_spec.rb

```diff
@@ -22,43 +22,87 @@
     Settings.zephir_dir = @old_z_dir
   end
 
-  it "generates the expected output" do
-    # One particular UC record from a day in August 2022
-    system("cp #{__dir__}/../data/000018677-20220807.json #{Settings.zephir_dir}/zephir_full_20220807.json")
+  context "with a full file" do
+    it "generates the expected output" do
+      # One particular UC record from a day in August 2022
+      system("cp #{__dir__}/../data/000018677-20220807.json #{Settings.zephir_dir}/zephir_full_20220807.json")
 
-    GenerateHathifile.new.run
+      GenerateHathifile.new.run
 
-    outfile = "#{Settings.hathifiles_dir}/hathi_full_20220808.txt.gz"
-    generated = Zlib::GzipReader.open(outfile).read
-    # The items as they were output on that day
-    expected = File.read("#{__dir__}/../data/000018677-20220808.tsv")
-    expect(generated).to eq(expected)
-  end
+      outfile = "#{Settings.hathifiles_dir}/hathi_full_20220808.txt.gz"
```

Comment:
All these tests are pretty much the same shape, and in duplicating them (full file vs. update file) we're sort of duplicating the duplication. Is there a helper function for the tests we could extract here?